### ISSUE TITLE
Reflection issue – Distribution issue

### ISSUE THREAD

**Issue by ousha2020** [ethereum/solidity#12307]
Hi Experts, 
I’m forking SafeMoon on test net, everything looks OK however the Smart Contract doesn’t distribute tokens for the holders, the contract is published on Test Net for your review, and I listed below all the steps I followed, it all went well except step # 5
I appreciate your feedback. 
1-	Contract Owner Launched the token (https://testnet.bscscan.com/address/0x1a2650AA65d96EbC7dd5854f0B9032615629F442#code) 
– Expected Behavior (Pass)
2-	Contract Owner created the pool with (Safemoonfork & BNB)
https://testnet.bscscan.com/tx/0xd5c4584dd7711afaf3f49ff8c280bcb22c253581e6f2566daf16cb18808fe956
– Expected Behavior (Pass)
3-	User 1 bought 30 token, received as 27 because of the 10 percent tax  (https://testnet.bscscan.com/tx/0xbb9f2518b441b86cbe34e227ea177157c060161ee8c947a504a0b265997ab9f1 )
– Expected Behavior (Pass)

4-	 User 2 bought 40 tokes, received as 36 because of the 10 percent tax  (https://testnet.bscscan.com/tx/0xc2372f808ce819c952b79de6f89c77e2ae19c0e8fb878c62ec70a30505f0c4fb) 
– Expected Behavior (Pass)

5-	User 1 should receive some tokens because he’s a holder, user 1 still have 27 Tokens only – Not Expected
– Not Expected Behavior (Fail)

**Comment by cameel** [comment]
Have you actually checked that the code is correct? Can you reproduce the same problem e.g. on Ganache or in Remix?

If this is really a bug in the compiler, it would be a serious problem in code generator and those are pretty rare in practice. Most often when we get reports like this it's actually a bug in the contract so we need to exclude that possibility first. Can you show a specific part of code that clearly didn't give a correct result based on the code and variable/parameter/state values?

**Comment by ousha2020** [comment]
Hi @cameel , thank you so much for your reply, I don't think it's a compiler issue. I copied the code from Safe Moon token (https://www.bscscan.com/address/0x8076c74c5e3f5852037f31ff0093eeb8c8add8d3#code) and I just changed the token name as you can see here ( https://testnet.bscscan.com/address/0x1a2650AA65d96EbC7dd5854f0B9032615629F442#code ), the code is working fine with the actual token Safe Moon and I was able to deploy the token using Remix, but it doesn't distribute tokens in my token, I think I'm missing something in how I should make this work! maybe it's related to the Pancake swap router or I should create a pool in a certain way! I don't know and thus, I posted these questions here to ask experts (like yourself) if you have experience in reflection tokens to let me know if I'm doing anything wrong!

**Comment by cameel** [comment]
Oh, in that case please try the [#soldity](https://gitter.im/ethereum/solidity) channel on gitter or [Ethereum Stack Exchange](https://ethereum.stackexchange.com). This bug tracker is only a place to report compiler bugs and submit feature requests.

**Comment by MRvinjak** [comment]
does you found the solution for this?

**Comment by ousha2020** [comment]
> does you found the solution for this?

Yes I found a solution, I used to make small transactions. thus, the reflection was too small to be viewed on the transaction, I increased the transaction size (E.G. %5 of the liquidity) and I saw the reflection coming, in a real-life scenario, the holder will see the election after multiple transactions (it will add up per transaction).
Let me know if this makes sense?
